Sources/TSCBasic/Process.swift

```diff
@@ -219,36 +227,36 @@ public final class Process: ObjectIdentifierProtocol {
     public private(set) var processID = ProcessID()
   #endif
 
-    /// If the subprocess has launched.
-    /// Note: This property is not protected by the serial queue because it is only mutated in `launch()`, which will be
-    /// called only once.
-    public private(set) var launched = false
+    // process execution mutable state
+    private var state: State = .idle
+    private let stateLock = Lock()
 
     /// The result of the process execution. Available after process is terminated.
+    /// This will block while the process is running, as such equivalent to `waitUntilExit`
```

Comment:
Based on the locking here it doesn't look as if it will block while the process is running, but rather return nil.  This is because I don't think `stateLock` is held all the time while the process is running (it's unlocked again on line 681).  So this comment is probably not accurate.

Reply:
(it does make sense that `stateLock` is only used to protect `state`, BTW, and isn't held for the duration of the process — if this property should block, could it instead just call `waitUntilExit`).

Reply:
it blocks while awaiting results (specifically `waitpid`), reworded accordingly

Reply:
Based on the current implementation, the semantics seem to be that it returns nil if called in `readingOutput` state (due to not holding the state lock during the `Thread.join()` calls) but blocks if called in `outputReady` state (due to holding the state lock during `waitpid()`).  So it depends on whether all the output from the process has been read yet.  Since the `result` property is deprecated and nothing seems to rely on it, I don't think that's a problem, but wanted to check if that matches the intent.

Reply:
short of removing this method, this is the only safe option I can think of. do you have other ideas we should explore?